### PR TITLE
Jerk parameter for auto mode

### PR DESCRIPTION
**Pull request by bresch**
Until now, `MPC_JERK_MIN` was used in `MPC_AUTO_MODE` 1 (limited-jerk trajectory). This was not specified in the description of the parameter and was not a good choice.

I created here a dedicated jerk parameter for auto mode. Also, the parameters specify in which manual/auto modes they are used and how.

### CODE REVIEW

File: src/modules/mc_pos_control/mc_pos_control_params.c [hunk 1]
```diff
@@ -454,6 +459,8 @@ PARAM_DEFINE_FLOAT(MPC_ACC_HOR, 5.0f);
 /**
  * Slow horizontal manual deceleration for manual mode
  *
+ * Note: This is only used when MPC_POS_MODE is set to 1 or 2.
```
Comment (MaEtUgR) [inline src/modules/mc_pos_control/mc_pos_control_params.c]: No smoothing in sport mode. See other comment.
```suggestion
 * Note: This is only used when MPC_POS_MODE is set to 1.
```

File: src/modules/mc_pos_control/mc_pos_control_params.c [hunk 2]
```diff
@@ -511,7 +518,7 @@ PARAM_DEFINE_FLOAT(MPC_ACC_DOWN_MAX, 10.0f);
  *
  * Setting this to the maximum value essentially disables the limit.
  *
- * Note: this is only used when MPC_POS_MODE is set to a smoothing mode.
+ * Note: This is only used when MPC_POS_MODE is set to a smoothing mode.
```
Comment (MaEtUgR) [inline src/modules/mc_pos_control/mc_pos_control_params.c]:
```suggestion
 * Note: This is only used when MPC_POS_MODE is set to a smoothing mode 1 or 3.
```

File: src/modules/mc_pos_control/mc_pos_control_params.c [hunk 3]
```diff
@@ -441,6 +444,8 @@ PARAM_DEFINE_FLOAT(MPC_ACC_HOR_MAX, 10.0f);
 /**
  * Acceleration for auto and for manual
  *
+ * Note: In manual, this parameter is only used in MPC_POS_MODE 1 and 2
```
Comment (MaEtUgR) [inline src/modules/mc_pos_control/mc_pos_control_params.c]:
```suggestion
 * Note: In manual, this parameter is only used in MPC_POS_MODE 1.
```

**Comment by MaEtUgR** [comment]
A related question to this pr: Why does FlightTaskManualPositionSmoothVel have the jerk_min parameter? I think it's unused: https://github.com/PX4/Firmware/blob/01fdd00c41dd5986e8b1faf0f735b406aed629b4/src/lib/FlightTasks/tasks/ManualPositionSmoothVel/FlightTaskManualPositionSmoothVel.hpp#L60

**Comment by bresch** [comment]
@MaEtUgR You are right, I planned to use it later for some dynamic constraint adjustments. I'll put it back when needed.
Please approve again :D

**Pull request merged**